lib/Doctrine/ODM/PHPCR/ReferenceManyCollection.php

```diff
@@ -150,9 +159,13 @@ public function getOriginalPaths()
                     $this->originalReferencePaths[] = $uow->getDocumentId($reference);
                 }
             } else {
-                $nodes = $this->dm->getPhpcrSession()->getNodesByIdentifier($this->referencedNodes);
-                foreach ($nodes as $node) {
-                    $this->originalReferencePaths[] = $node->getPath();
+                if ($this->referenceType == 'uuid') {
```

Comment:
same here. and maybe switch the cases here or above to have the same order in both.